### DISCUSSION TITLE
kwargs from the lightgbm predict are not available with the LightGBMLSS.predict

### DISCUSSION THREAD

**Discussion by EchoDel** [StatMixedML/LightGBMLSS#32] [ToDo]
For model governance reason I want to be able to run tree by tree using the start_iteration and num_iteration arguments to see if there is a number of over fitted trees towards the end of the training.

Would it be possible to add in the **kwargs to the predict all the way through to the lightgbm.predict.

**Comment by StatMixedML** [comment]
@EchoDel Thanks for raising this issue.

I would rather suggest to add an additional function that allows to return the trained `lgb.Booster` so that you can use it for further analysis. What do you think?

**Comment by EchoDel** [comment]
I'm looking for the parameters after the transformations and the probabilities so to avoid duplicating the code in the distribution class I would prefer either specifying the kwargs or being able to pass the raw predictions in to the distribution class to apply the transformations.

**Comment by StatMixedML** [comment]
Yeah, you are probably correct. Would be good to have the kwargs available. Not sure though when I find the time to look into it.

**Comment by EchoDel** [comment]
So I can work through it and submit and PR which you can merge after the tests have been added but thinking about if there is a better way to do this.
Does it make sense to change the DistributionClass.predict_dist to take a predictions argument instead of the booster and test set, then moving the booster.predict into the LightGBMLSS.predict?
Again I can work through this and submit a PR if you are happy with this change.

**Comment by StatMixedML** [comment]
Sorry was busy with the recent releases. Yes makes sense, thank you. Happy if you create a first draft via a PR
